redis-sink/src/main/java/org/springframework/cloud/stream/module/redis/sink/RedisSinkAutoConfiguration.java

```diff
@@ -0,0 +1,66 @@
+package org.springframework.cloud.stream.module.redis.sink;
+
+import org.apache.commons.pool2.impl.GenericObjectPool;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingClass;
+import org.springframework.boot.autoconfigure.redis.RedisAutoConfiguration;
+import org.springframework.boot.autoconfigure.redis.RedisProperties;
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.data.redis.connection.RedisConnectionFactory;
+import org.springframework.data.redis.connection.jedis.JedisConnection;
+import org.springframework.data.redis.connection.jedis.JedisConnectionFactory;
+import org.springframework.data.redis.core.RedisOperations;
+import org.springframework.integration.redis.outbound.RedisPublishingMessageHandler;
+import org.springframework.integration.redis.outbound.RedisQueueOutboundChannelAdapter;
+import org.springframework.integration.redis.outbound.RedisStoreWritingMessageHandler;
+import org.springframework.messaging.MessageHandler;
+import redis.clients.jedis.Jedis;
+
+import java.net.UnknownHostException;
+
+/**
+ * Creates a dedicated RedisConnectionFactory different from the one spring-boot autoconfigure
+ * section that the bindings use in spring-cloud-stream.
+ *
+ * The configuration prefix is "spring.cloud.stream.module.redis" and contains the standard
+ * properties to configure a redis connection, host, port, etc as well as the additional properties for
+ * the sink, queue, key, etc.
+ *
+ * @author Eric Bottard
+ * @author Mark Pollack
+ */
+@Configuration
+@ConditionalOnClass({ JedisConnection.class, RedisOperations.class, Jedis.class })
```

Comment:
We don't need `@ConditionalOnClass` this as the RedisSink has a compile time dependency on these classes.